scripts/zones/Waughroon_Shrine/mobs/Princess_Jelly.lua

```diff
@@ -0,0 +1,185 @@
+-----------------------------------
+-- Area: Waughroon Shrine
+--  MOB: princess_jelly
+-- Author: Setzor (ported to LSB by dallano)
+-----------------------------------
+require("scripts/globals/pathfind")
+require("scripts/globals/utils")
+-----------------------------------
+local princesses =
+{
+    {17367174, 17367175, 17367176, 17367177, 17367178, 17367179, 17367180, 17367181},
+    {17367184, 17367185, 17367186, 17367187, 17367188, 17367189, 17367190, 17367191},
+    {17367194, 17367195, 17367196, 17367197, 17367198, 17367199, 17367200, 17367201},
+}
+local elementalSpells =
+{
+    {235,144}, -- burn, fire
+    {240,169}, -- drown, water
+    {239,164}, -- shock, thunder
+    {238,159}, -- rasp, stone
+    {237,154}, -- choke, aero
+    {236,149}, -- frost, blizard
+    { 23, 28}, -- dia, banish
+    {230,245}, -- bio, drain
+}
+local centers =
+{
+    {-177.5,60,-142},
+    {22.5,   0,  18},
+    {222.5,-60, 138},
+}
+local queens =
+{
+    17367173,
+    17367183,
+    17367193,
+}
+
+local entity = {}
+
+entity.onMobInitialize = function(mob)
+    mob:setMobMod(xi.mobMod.MAGIC_COOL, 40)
+end
+
+entity.onMobSpawn = function(mob)
+    mob:setSpeed(1)
+    mob:setUnkillable(false)
+    mob:setLocalVar('mobElement', math.random(1,8))
+    local element = mob:getLocalVar('mobElement')
+
+    if element == 1 then
+        mob:addMod(xi.mod.WATER_MEVA, -250)
+    elseif element == 2 then
+        mob:addMod(xi.mod.THUNDER_MEVA, -250)
+    elseif element == 3 then
+        mob:addMod(xi.mod.EARTH_MEVA, -250)
+    elseif element == 4 then
+        mob:addMod(xi.mod.WIND_MEVA, -250)
+    elseif element == 5 then
+        mob:addMod(xi.mod.ICE_MEVA, -250)
+    elseif element == 6 then
+        mob:addMod(xi.mod.FIRE_MEVA, -250)
+    elseif element == 7 then
+        mob:addMod(xi.mod.DARK_MEVA, -250)
+    else
+        mob:addMod(xi.mod.LIGHT_MEVA, -250)
+    end
+end
+
+local function getDistanceFromCenter(bfNum,mob)
+    local pos = mob:getPos()
+
+    local centerX = centers[bfNum][1]
```

Comment:
Would suggest just pulling the array as a single value, or assigning this directly in difX/Y/Z.  Seems a bit excessive